### PR TITLE
Merge hazardlib inside the Engine repo

### PR DESCRIPTION
**Pull request by daniviga**
This PR wants to merge `oq-hazardlib` inside the `oq-engine` repo.

Only one `deb` and one `rpm` package are now generated (`python-oq-engine`).

## TODO during/after the merge ##

- [x] freeze `oq-hazardlib` and do the last sync
- [x] migrate Jenkins CI Linux jobs (see https://ci.openquake.org/job/zdevel_oq-engine_merge/)
- [x] update Jenkins CI Mac and Windows jobs
- [x] update cluster deployment scripts and console
- [x] update all the installers, docker images and VM scripts (https://github.com/gem/oq-builders/pull/61)
- [x] update pypi uploader job
- [ ] update the internal documentation
- [ ] remove old hazardlib package from nightly ppa and unstable COPR
- [ ] make an announcement on `openquake-users` and staff

## Tests ##
CI: https://ci.openquake.org/job/zdevel_oq-engine_merge/8/
RPM: https://ci.openquake.org/job/builders/job/rpm-builder_merge/1/
Travis: https://travis-ci.org/gem/oq-engine/builds/239849280

A note about Travis: to avoid issue with the log size and to gain speed, the Travis configuration has been splitted in multiple jobs:

screenshot from 2017-06-05 18-33-01

See https://github.com/gem/oq-engine/blob/hazardlib/.travis.yml

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,27 +5,45 @@ language: python
...
+        - nosetests --with-doctest -xvs -a'!slow' openquake.commonlib
+        - nosetests --with-doctest -xvs -a'!slow' openquake.commands
+  - stage: tests
+    env: HMTK
+    script:
+        - nosetests --with-doctest -xvs -a'!slow' openquake.hmtk
+  - stage: demos
+    env: DEMOS
+    script: 
+        - bin/run-demos.sh
```
Comment (micheles) [inline .travis.yml]: There are too many stages. I would keep only three:

TESTS/HAZARDLIB:
   baselib, hazardlib, hmtk
TESTS/ENGINE:
   engine,server,calculators,risklib,commonlib,commands 
DEMOS:
     as it is

File: doc/development.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
+# Installing the OpenQuake Hazardlib for development
+
```
Comment (micheles) [inline doc/development.md]: I guess this should be renamed Installing the `OpenQuake Engine for development`. There should be a single document both for engine and hazardlib.

File: doc/sphinx/Makefile_old [hunk 3]
```diff
@@ -0,0 +1,153 @@
+# Makefile for Sphinx documentation
```
Comment (micheles) [inline doc/sphinx/Makefile_old]: I guess this file should be removed

File: doc/sphinx/make.bat_old [hunk 4]
```diff
@@ -0,0 +1,190 @@
+@ECHO OFF
```
Comment (micheles) [inline doc/sphinx/make.bat_old]: Also this should be removed

File: doc/development.md (hunk 2 above)
Comment (daniviga) [inline doc/development.md]: This has been removed, was left during the merge

File: doc/sphinx/Makefile_old (hunk 3 above)
Comment (daniviga) [inline doc/sphinx/Makefile_old]: Done

File: doc/sphinx/make.bat_old (hunk 4 above)
Comment (daniviga) [inline doc/sphinx/make.bat_old]: Done

File: .travis.yml (hunk 1 above)
Comment (daniviga) [inline .travis.yml]: Done

**Comment by mmpagani** [comment]
LGTM

**Comment by micheles** [comment]
LGTM

**Comment by daniviga** [comment]
I would propose to do the merge tomorrow early morning, sounds good?

**Comment by micheles** [comment]
OK

**Pull request merged**